server/services/message-service.js

```diff
@@ -1,38 +1,65 @@
-const amqp = require('amqplib/callback_api')
-const scheduleService = require('./schedule-service')
+const rheaPromise = require('rhea-promise')
 const config = require('../config')
+const scheduleService = require('./schedule-service')
 
 module.exports = {
-  receiveSchedule: function () {
-    const messageQueue = config.messageQueue
-    amqp.connect(messageQueue, function (err, conn) {
-      if (err) {
-        console.log(err)
-      } else {
-        conn.createChannel(function (err, ch) {
-          if (err) {
-            console.log(err)
-          } else {
-            const scheduleQueue = 'schedule'
-            ch.assertQueue(scheduleQueue, { durable: false })
-            console.log('waiting for new claims')
+  setupReceivers: async function (messageQueueOptions) {
+    const connectionOptions = configureMQ(messageQueueOptions || config.messageQueue)
+    const connection = new rheaPromise.Connection(connectionOptions)
 
-            ch.consume(scheduleQueue, function (msg) {
-              console.log(`claim received for scheduling - ${msg.content.toString()}`)
-              scheduleService.create(JSON.parse(msg.content))
-            }, { noAck: true })
+    try {
+      console.log('opening connection')
+      await connection.open()
+    } catch (err) {
+      console.log(`unable to connect to message queue ${err}`)
+    }
 
-            const valueQueue = 'value'
-            ch.assertQueue(valueQueue, { durable: false })
-            console.log('waiting for new calculation')
+    Promise.all([
+      setupReceiver(connection, 'payment-service-schedule', config.messageQueue.scheduleAddress),
+      setupReceiver(connection, 'payment-service-value', config.messageQueue.valueAddress)
+    ])

```

Comment:
Nothing is done with the result of the Promise.all, there needs to be an `await` and additional error handing to prevent unhandled promise rejections if setting up a receiver fails.

Reply:
I've updated this, can you take another look at it.  I've not really worked much with promises before so appreciate any advice you can give me.